api/src/services/server.ts

```diff
@@ -307,9 +315,16 @@ export class ServerService {
 				const startTime = performance.now();
 
 				try {
-					await disk.write(`health-${checkID}`, Readable.from(['check']));
-					await disk.read(`health-${checkID}`);
-					await disk.delete(`health-${checkID}`);
+					await disk.write(`health-${checkID}`, Readable.from(['check'])).catch((err) => {
+						throw err;
+					});
+					await sleep(500); // 500 ms wait for disk to write
```

Comment:
What error situation does this wait fix and could this still fail if i manage to make the write take longer than 500ms? 🤔

Reply:
if the wait is longer than 500ms, we get into healthcheck warning thresholds about healthcheck taking too long, all the sleep is doing is giving a slight pause between the write and the read to make sure any streams are closed, etc, without it we can still get the crash if the healthchecks are called too close together

Reply:
removed sleep in favor of listeners in the adapter